### ISSUE TITLE
Using w/ TS

### ISSUE THREAD

**Issue by bfricka** [theKashey/jsx-compress-loader#2]
I'm very sorry to post this here, but there is so little information out there that I could find. Is there some way to use this w/ TypeScript? I have an ejected CRA app, and I've tried all sorts of configurations.

None of them work. In fact, in no cases have I even seen any of my application files even show up in the transform function. All of the files that come through the transform function are library files.

The only sort of "weird" thing I use in my config, is [@ts-tools/webpack-loader](https://github.com/AviVahl/ts-tools). I use this b/c it's the only loader I've tried that can be configured to work well w/ a multiple TS project monorepo.

I've tried changing the order of the loaders. I've tried pulling all the JS/TS loaders out of the `oneOf` block. In the end, my compressed output always has `r.a.createElement`.

If you have any ideas, I would very much appreciate it.

**Comment by theKashey** [comment]
It is working well with TS, just double checked. There is only one requirement - it shall be placed just after `ts-loader`(or `awesome-ts-loader`) - any minifications of modifications would break the magic.
```
config.module.rules.push({
    test: /\.(ts|tsx)$/,
    use: [
      {
        loader: require.resolve('jsx-compress-loader'),
      },
      {
        loader: require.resolve('awesome-typescript-loader'),
      },
    ],
  });
```

You may send me a piece of your code (compiled in dev mode), and I'll check why it's not working.

**Comment by bfricka** [comment]
I actually tried that before and kept getting an error:

```
TypeError: aSourceMapConsumer.eachMapping is not a function
```
I went through and searched for `aSourceMapConsumer.eachMapping` and found it associated w/ `@babel/generator` as well as the peer dependency `source-map` required for this module. They seemed to be a bit at odds, so on a hunch, I removed the `source-map` peer dependency I had installed for this module, and everything worked.

Thanks.
